src/main/java/jenkins/advancedqueue/PriorityConfigurationPlaceholderTaskHelper.java

```diff
@@ -0,0 +1,29 @@
+package jenkins.advancedqueue;
+
+
+import hudson.model.Job;
+import hudson.model.Queue;
+import jenkins.advancedqueue.sorter.ItemInfo;
+import jenkins.advancedqueue.sorter.QueueItemCache;
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.workflow.support.steps.ExecutorStepExecution;
+
+class PriorityConfigurationPlaceholderTaskHelper {
+
+    boolean isPlaceholderTask(Queue.Task task) {
+        return isPlaceholderTaskUsed() && task instanceof  ExecutorStepExecution.PlaceholderTask;
+    }
+
+    PriorityConfigurationCallback getPriority(ExecutorStepExecution.PlaceholderTask task, PriorityConfigurationCallback priorityCallback) {
+        Job<?, ?> job = (Job<?, ?>) task.getOwnerTask();
+        ItemInfo itemInfo = QueueItemCache.get().getItem(job.getName());
+        itemInfo.getPriority();
+        priorityCallback.setPrioritySelection(itemInfo.getPriority());
+        return priorityCallback;
+    }
+
+    static boolean isPlaceholderTaskUsed() {
+        return Jenkins.getInstance().getPlugin("workflow-durable-task-step") != null;
```

Comment:
After checking the code I'm not 100% sure about it. The plugin may be registered, but not active yet. In such case classloader won't be able to find the class. This method will return true, but then isPlaceholderTask() will fail with ClassNotFoundException. Need to reproduce it, but it's what I've seen on old core versions at least